pkg/productsModel/products.go

```diff
@@ -0,0 +1,78 @@
+package productsModel
+
+import (
+	"context"
+	"github.com/appuio/appuio-cloud-reporting/pkg/db"
+	"github.com/jmoiron/sqlx"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/tokenMatcher"
+	"time"
+)
+
+func GetBySource(ctx context.Context, tx *sqlx.Tx, source string) (*db.Product, error) {
+	var products []db.Product
+	err := sqlx.SelectContext(ctx, tx, &products, `SELECT products.* FROM products WHERE source = $1`, source)
+	if err != nil {
+		return nil, err
+	}
+	if len(products) == 0 {
+		return nil, nil
+	}
+	return &products[0], nil
+}
+
+func getBySourceQueryAndTime(ctx context.Context, tx *sqlx.Tx, sourceQuery string, timestamp time.Time) ([]db.Product, error) {
+	var products []db.Product
+	err := sqlx.SelectContext(ctx, tx, &products,
+		`SELECT products.* FROM products 
+                  WHERE (source = $1 OR source LIKE $2)
+                  AND during @> $3::timestamptz`,
+		sourceQuery, sourceQuery+":%", timestamp)
+	if err != nil {
+		return nil, err
```

Comment:
If you didn't know: you can enrich the returned error by wrapping it using `fmt.Errorf()`. I believe this would add a stack trace, at least a trace until this line. And it would produce a more human friendly error. https://pkg.go.dev/fmt#Errorf

Suggestion applies to other locations as well, but I'm not gonna mark them all.
```suggestion
		return nil, fmt.Errorf("cannot get products by source %q and timestamp %s: %w", sourceQuery, timestamp, err)
```

Reply:
Done.